commons/zenoh-buffers/src/zslice.rs

```diff
@@ -334,10 +286,57 @@ impl Buffer for &mut ZSlice {
 
 // SplitBuffer
 impl SplitBuffer for ZSlice {
-    type Slices<'a> = option::IntoIter<&'a [u8]>;
+    type Slices<'a> = iter::Once<&'a [u8]>;
 
     fn slices(&self) -> Self::Slices<'_> {
-        Some(self.as_slice()).into_iter()
+        iter::once(self.as_slice())
+    }
+}
+
+#[derive(Debug)]
+pub struct ZSliceWriter<'a> {
```

Comment:
`ZSlice` has been since the beginning a `read-only` type and we have avoided adding any writing capability to the `ZSlice` till now. Could it just been moved to `ZBuf` and kept as internal type? What's the benefit of having it public?

Reply:
A mistake, it could be `pub(crate)` indeed.